### PR TITLE
[Go] Prevent naming collisions between model properties and generated getter/setter methods

### PR DESCRIPTION
**Pull request by DouglasDwyer**
This PR adds a small fix for naming collisions in Go between generated struct fields and getter/setter method names. When using the Go generator, specifications that include model properties like `Area` and `HasArea` will cause code to be generated with compilation errors. This occurs because the `HasArea` field on the generated model has the same name as the `HasArea()` function for checking whether the model contains an `Area` property. This change adds a conflict resolution method which automatically renames model fields by appending underscore characters (`_`) to them until no more naming conflicts occur (without changing the underlying JSON names).

Here is a simple example of a specification which causes issues:

```yml
openapi: 3.0.1
info:
  title: Test REST API
  description: ''
  termsOfService: ''
  version: 0.0.0
paths: {}
components:
  schemas:
    Broken:
      type: object
      properties:
        hasMass:
          type: boolean
        mass:
          type: array
          items:
            type: number
            format: double
```

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.0.1) (patch release), `6.1.x` (breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request. @antihax @grokify @kemokemo @jirikuncar @ph4r5h4d

### CODE REVIEW

File: docs/generators/fsharp-functions.md [hunk 1]
```diff
@@ -34,7 +34,7 @@ These options may be applied as additional-properties (cli) or configOptions (pl
 |prependFormOrBodyParameters|Add form or body parameters to the beginning of the parameter list.| |false|
 |sortModelPropertiesByRequiredFlag|Sort model properties to place required parameters before optional parameters.| |true|
 |sortParamsByRequiredFlag|Sort method arguments to place required parameters before optional parameters.| |true|
-|sourceFolder|source folder for generated code| |OpenAPI/src|
+|sourceFolder|source folder for generated code| |OpenAPI\src|
```
Comment (wing328) [inline docs/generators/fsharp-functions.md:37]: Thanks for the PR. Are you using Git Bash when updating the samples?

I've never been able to repeat the issue in my Windows laptop so I would appreciate if you can share more about your development environment/setting.

Comment (DouglasDwyer) [reply docs/generators/fsharp-functions.md:37]: Indeed, I am on Windows 11 and using Git Bash to update the samples. I just followed the instructions in the pull request template, but according to my understanding, none of the samples should actually be affected by my edits. I would be happy to try regenerating the samples again and see whether the backslashes still appear. I can also try generating the samples using a Linux environment, which will probably alleviate the issue. Please let me know what needs to happen in order for this pull request to move forward. Thanks!

Comment (DouglasDwyer) [reply docs/generators/fsharp-functions.md]: I went ahead and regenerated the samples on Linux. This seems to have fixed the file separator issue, so I hope that we can move forward with the changes :)

Comment (wing328) [reply docs/generators/fsharp-functions.md]: 👍 

will try to get it merged in the next few days.

**Comment by wing328** [comment]
we've added name mapping options to avoid name collision: https://github.com/OpenAPITools/openapi-generator/blob/master/docs/customization.md#name-mapping

please give it a try with the latest master.

closing this one for the time being, happy to reopen if needed.

**Pull request closed without merging**